web-app/src/app/context/RemoteConfigProvider.spec.tsx

```diff
@@ -0,0 +1,66 @@
+import { type BypassConfig } from '../interface/RemoteConfig';
+import { userHasBypass } from './RemoteConfigProvider';
+jest.mock('firebase/compat/app', () => ({
+  initializeApp: jest.fn(),
+  remoteConfig: jest.fn(() => ({
+    settings: { minimumFetchIntervalMillis: 3600000 },
+  })),
+}));
+describe('userHasBypass', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should return false if userEmail is null', () => {
+    const byPassConfig: BypassConfig = { regex: ['.*@example.com'] };
```

Comment:
Let's add a test with a regex expression that has more strict email verification like: `[^@]+@example\.com`